### PR TITLE
Simulate updates

### PR DESCRIPTION
**Pull request by mvdan**

### CODE REVIEW

File: simulate/Main.hs [hunk 1]
```diff
@@ -42,19 +33,18 @@ instance Show DynSignal where show (Signal i) = [chr (ord 'A' + i)]
...
     "data DynSignal = Signal Int deriving Eq",
-    "instance Show DynSignal where show (Signal i) = [chr (ord 'A' + i)]",
     "signs = [Signal i | i <- [0.." ++ show (num-1) ++ "]]",
     "apply c = c " ++ unwords ["(signs !! " ++ show i ++ ")" | i <- [0..num-1]]]
 
 writeTmpFile :: [String] -> IO ()
 writeTmpFile ls =
     writeFile tmpModuleFile $ unlines withModule
-      where withModule = ("module " ++ tmpModuleName ++ " where") : ls
+      where withModule = ("module Helper where") : ls
```
Comment (snowleopard) [inline simulate/Main.hs]: Redundant brackets? Just a minor nitpick :)

Comment (mvdan) [inline simulate/Main.hs]: Correct! They became redundant with the removal of the two `++`.

**Comment by snowleopard** [comment]
The rest seems fine. Let me know when ready.

**Comment by mvdan** [comment]
Already fixed the redundant brackets, we can merge on green as far as I'm concerned.

**Comment by snowleopard** [comment]
:+1: Will do!

**Pull request merged**